app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,49 @@
+class HallsController < ApplicationController
+before_action :set_hall, only: %i[ show edit update destroy ]
+
+  def index
+    @halls = Hall.all
+  end
+
+  def new
+    @hall = Hall.new
+    @hall_path = halls_path
+  end
+
+  def create
+    @hall = Hall.new(hall_params)
+
+    if @hall.save
+      redirect_to halls_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+      @hall_path = hall_path(@hall)
```

Comment:
You have a problem with indentation here, did you run Rubocop or anything else like that before you committed those changes?

Reply:
No, good catch!